### PR TITLE
compute: support large  disk

### PR DESCRIPTION
**Pull request by yugangw-msft**
The main thing is the `upload` option on create a disk that you can upload blobs into it later   

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py [hunk 1]
```diff
@@ -151,11 +151,11 @@ def _get_sku_object(cmd, sku):
     return sku
 
 
-def _grant_access(cmd, resource_group_name, name, duration_in_seconds, is_disk):
+def _grant_access(cmd, resource_group_name, name, duration_in_seconds, is_disk, access_level):
     AccessLevel = cmd.get_models('AccessLevel')
     client = _compute_client_factory(cmd.cli_ctx)
     op = client.disks if is_disk else client.snapshots
-    return op.grant_access(resource_group_name, name, AccessLevel.read, duration_in_seconds)
+    return op.grant_access(resource_group_name, name, access_level or AccessLevel.read, duration_in_seconds)
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py]: i will add example commands to show the whole e2e including the storage commands

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -68,6 +68,11 @@ def load_arguments(self, _):
             c.argument('source_storage_account_id', help='used when source blob is in a different subscription')
             c.argument('size_gb', options_list=['--size-gb', '-z'], help='size in GB. Max size: 4095 GB (certain preview disks can be larger).', type=int)
             c.argument('duration_in_seconds', help='Time duration in seconds until the SAS access expires', type=int)
+            if self.supported_api_version(min_api='2018-09-30', operation_group='disks'):
+                c.argument('access_level', arg_type=get_enum_type(['Read', 'Write']), default='Read', help='access level')
+                c.argument('for_upload', arg_type=get_three_state_flag(),
+                           help='create the disk for uploading blobs later through storage commands, run "az disk/snpashot grant-access --access-level Write" to get required SAS token')
```
Comment (adewaleo) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]:
```suggestion
                           help='Create the {0} for uploading blobs to later on through storage commands. Run "az {0} grant-access --access-level Write" to retrieve the {0}'s SAS token.'.format(scope))
```

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 3]
```diff
@@ -676,6 +676,23 @@ def test_vm_managed_disk(self, resource_group):
                      self.check('storageProfile.osDisk.caching', 'ReadWrite')
                  ])
 
+    @ResourceGroupPreparer(name_prefix='cli_test_large_disk')
+    def test_vm_larget_disk(self, resource_group):
```
Comment (adewaleo) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py]: typo:

```suggestion
    def test_vm_large_disk(self, resource_group):
```

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 4]
```diff
@@ -676,6 +676,23 @@ def test_vm_managed_disk(self, resource_group):
...
+        self.cmd('disk create -g {rg} -n {disk} --size-gb 1 --hyper-v-generation V2 --for-upload', checks=[
+            self.check('hyperVgeneration', "V2")
+        ])
+        self.cmd('disk grant-access -g {rg} -n {disk} --access-level Write --duration-in-seconds 3600')
+        # To connfirm with compute team that snapshot indeed have --hyper-v-generation support
+        # self.cmd('disk create -g {rg} -n {disk2} --size-gb 1')
+        # self.cmd('snapshot create -g {rg} -n {snapshot} --source {disk2} --hyper-v-generation V2', checks=[
+        #     self.check('hyperVgeneration', "V2")
+        # ])
+
```
Comment (adewaleo) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py]: Will a simple e2e test of being able to upload to the disk be overkill or helpful?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 5]
```diff
@@ -676,6 +676,23 @@ def test_vm_managed_disk(self, resource_group):
...
+        self.kwargs.update({
+            'disk': 'd1',
+            'disk2': 'd2',
+            'snapshot': 's1'
+        })
+        self.cmd('disk create -g {rg} -n {disk} --size-gb 1 --hyper-v-generation V2 --for-upload', checks=[
+            self.check('hyperVgeneration', "V2")
+        ])
+        self.cmd('disk grant-access -g {rg} -n {disk} --access-level Write --duration-in-seconds 3600')
+        # To connfirm with compute team that snapshot indeed have --hyper-v-generation support
```
Comment (adewaleo) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py]: @yugangw-msft should this commented-out code be here?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 6]
```diff
@@ -1553,7 +1570,7 @@ def test_vm_ultra_ssd_storage_sku(self, resource_group):
         self.cmd('disk create -g {rg} -n {disk1} --size-gb 4 --sku UltraSSD_LRS --disk-iops-read-write 500 --disk-mbps-read-write 8 --zone 3')
         self.cmd('disk show -g {rg} -n {disk1}')
         self.cmd('disk create -g {rg} -n {disk2} --size-gb 4 --sku UltraSSD_LRS')
-        self.cmd('vm create -g {rg} -n {vm} --admin-username admin123 --admin-password testPassword0 --image debian --storage-sku UltraSSD_LRS --data-disk-sizes-gb 4 --zone 3 --location eastus2')
+        self.cmd('vm create -g {rg} -n {vm} --admin-username admin123 --size Standard_D2s_v3 --admin-password testPassword0 --image debian --storage-sku UltraSSD_LRS --data-disk-sizes-gb 4 --zone 3 --location eastus2')
```
Comment (adewaleo) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py]: Why this change? Is this the cause of test flakiness, in our live tests?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/recordings/test_vm_larget_disk.yaml [hunk 7]
```diff
@@ -0,0 +1,1044 @@
+interactions:
+- request:
```
Comment (adewaleo) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/recordings/test_vm_larget_disk.yaml]: We will need to delete this file after renaming the test. Should be: test_vm_large_disk.yaml

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py (hunk 3 above)
Comment (adewaleo) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py]: typo:

```suggestion
    def test_vm_large_disk(self, resource_group):
```

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py (hunk 5 above)
Comment (adewaleo) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py]: should this commented-out code be here?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py (hunk 6 above)
Comment (adewaleo) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py]: Is this to fix flakey live tests? Asking out of curiosity.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py (hunk 4 above)
Comment (adewaleo) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py]: Will an end-to-end test where we upload some data to a large disk be helpful or overkill?

Comment (yugangw-msft) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py]: It is a bit overkill to create vhd and upload it, but I have uncommented the test steps here fpr end to end of capturing from a disk should work.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py (hunk 6 above)
Comment (yugangw-msft) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py]: yes, the default vm size of ds1_v2 no longer supports the ultra ssd

**Comment by adewaleo** [comment]
This pr closes #8774
